sources/packages/web/src/views/institution/locations/programs/LocationProgramOffering.vue

```diff
@@ -127,6 +158,13 @@ export default {
       }
       return "Add Study Period";
     });
+    const hasExistingApplication = computed(
+      () =>
+        initialData.value.hasExistingApplication &&
+        isInstitutionUser.value &&
+        initialData.value.offeringStatus === OfferingStatus.Approved,
```

Comment:
For my understanding, can an application be considered as an existing application only when it is approved or does it also include a pending application that may or may not get approved eventually?

Reply:
When an offering is validated to check if it has any application linked to it, we check the student_assessments table to see if the given offering is present in student_assessments table at all. 
What it means is, if there was at least one assessment started with (creation of workflow instance) given offering for any student application in the present or past, then it is considered to be linked to a student application. And in such case institution user will not be able to update any of offering fields except offering name and they need to request for change to update the offering further.